backend/pipelined/handle.go

```diff
@@ -110,12 +111,21 @@ func (p *Pipelined) expandHandlers(ctx context.Context, handlers []string, level
 
 		if handler == nil {
 			if err != nil {
-				logger.WithError(err).Error("pipelined failed to retrieve a handler")
+				(logger.
+					WithFields(logrus.Fields{"handler": handlerName}).
+					WithError(err).
+					Error("pipelined failed to retrieve a handler", handlerName))
```

Comment:
Thank you for adding the `handlerName` as a logrus field! I hate to be logger nanny here, but we can probably remove it from the message in `Error()` since it's already a field.